lib/blacklight/access_controls/enforcement.rb

```diff
@@ -25,72 +29,61 @@ module Enforcement
-        user_access_filters = []
-
-        # Grant access based on user id & group
-        solr_access_filters_logic.each do |method_name|
-          user_access_filters += send(method_name, permission_types, ability)
-        end
-        user_access_filters
-      end
-
-      #
-      # Solr query modifications
-      #
-
-      # Set solr_parameters to enforce appropriate permissions
-      # * Applies a lucene query to the solr :q parameter for gated discovery
-      # * Uses public_qt search handler if user does not have "read" permissions
-      # @param solr_parameters the current solr parameters
-      def add_access_controls_to_solr_params(solr_parameters)
-        apply_gated_discovery(solr_parameters)
-      end
-
       # Which permission levels (logical OR) will grant you the ability to discover documents in a search.
-      # Override this method if you want it to be something other than the default
+      # Override this method if you want it to be something other than the default, or hit the setter
       def discovery_permissions
         @discovery_permissions ||= %w(discover read)
       end
 
-      def discovery_permissions=(permissions)
-        @discovery_permissions = permissions
+      protected
+
+      # Grant access based on user id & group
+      # @return [Array{Array{String}}]
+      def gated_discovery_filters(permission_types = discovery_permissions, ability = current_ability)
+        solr_access_filters_logic.map { |method| send(method, permission_types, ability).reject(&:blank?) }
       end
 
-      # Controller before filter that sets up access-controlled lucene query in order to provide gated discovery behavior
-      # @param solr_parameters the current solr parameters
+      ### Solr query modifications
+
+      # Controller before_filter that sets up access-controlled lucene query to provide gated discovery behavior.
+      # Set solr_parameters to enforce appropriate permissions.
+      # @param [Hash{Object}] solr_parameters the current solr parameters, to be modified herein!
+      # @note Applies a lucene filter query to the solr :fq parameter for gated discovery.
       def apply_gated_discovery(solr_parameters)
         solr_parameters[:fq] ||= []
-        solr_parameters[:fq] << gated_discovery_filters.join(' OR ')
+        solr_parameters[:fq] << gated_discovery_filters.reject(&:blank?).join(' OR ')
         Rails.logger.debug("Solr parameters: #{solr_parameters.inspect}")
       end
 
+      # For groups
+      # @return [Array{String}] values are lucence syntax term queries suitable for :fq
+      # @example
+      #   [ "({!terms f=discover_access_group_ssim}public,faculty,africana-faculty,registered)",
+      #     "({!terms f=read_access_group_ssim}public,faculty,africana-faculty,registered)" ]
       def apply_group_permissions(permission_types, ability = current_ability)
-        # for groups
+        groups = ability.user_groups
+        return [] if groups.empty?
```

Comment:
@jcoyne Do you consider this incorrect?  If so, then I would consider the parallel case to `apply_group_permissions` in `apply_user_permissions` also incorrect:
```ruby
return [] unless user && user.user_key.present?
```
While I cleaned up the form here, that behavior was pre-existing.  No user_key, no filters.  That was the scenario that actually had tests backing it, so I regarded it as more intentional:
https://github.com/projectblacklight/blacklight-access_controls/blob/eeac826692c056d55db8ef94a9a12cfa08578a92/spec/unit/enforcement_spec.rb#L134-L147

Happy to adapt if there is something more formal to work from, but this is my best understanding of the intended behavior.